mathics/builtin/lists.py

```diff
@@ -2548,11 +2548,35 @@ def parts():
             e.message(evaluation)
 
 
+class Insert(Builtin):
+    """
+    <dl>
+      <dt>'Insert[$list$, $elem$, $n$]'
+      <dd>inserts $elem$ at position $n$ in $list$. When $n$ is negative, the position is counted from the end.
+    </dl>
+
+    >> Insert[{a,b,c,d,e}, x, 3]
+     = {a, b, x, c, d, e}
+
+    >> Insert[{a,b,c,d,e}, x, -2]
+     = {a, b, c, d, x, e}
+    """
+    def apply(self, expr, elem, n, evaluation):
+        'Insert[expr_List, elem_, n_Integer]'
+
+        py_n = n.to_python()
```

Comment:
We could probably use `n.get_int_value()` in here (it's a bit safer than `to_python`, but it isn't strictly necessary because `n` is guaranteed to be an integer).

Reply:
I thiink we should be using `to_python()` for a number of reasons:

- it requires less thought to get it right. (How many times in PR's have I seen the comment that there is `get_int_value()` instead of ... this is also an indicator that something is not simple )
- it is more robust when thing change.